### ISSUE TITLE
[Feature Request] HTML export with truly standalone HTML

### ISSUE THREAD

**Issue by tatoosh** [BoostIO/Boostnote#1772]
The HTML export is nice but not the best:

- images are not included - why not using inline image?

```html
<p data-line="6"><img src=":storage%5C0.04i5dwoqj2x6.png" alt="0.04i5dwoqj2x6" /></p>
```

- CSS is referenced to css folder

```html
  <link rel="stylesheet" href="css/ambiance-mobile.css"><link rel="stylesheet" href="css/katex.min.css"><link rel="stylesheet" href="css/codemirror.css">
```

- fonts are referenced to local filesystem

```html
  font-family: 'Lato';
  src: url('file://C:\Users\ta\AppData\Local\boost\app-0.11.3\resources\app/resources/fonts/Lato-Black.woff2') format('woff2'), /* Modern Browsers */
       url('file://C:\Users\ta\AppData\Local\boost\app-0.11.3\resources\app/resources/fonts/Lato-Black.woff') format('woff'), /* Modern Browsers */
       url('file://C:\Users\ta\AppData\Local\boost\app-0.11.3\resources\app/resources/fonts/Lato-Black.ttf') format('truetype');
  font-style: normal;
  font-weight: 700;
  text-rendering: optimizeLegibility;
```

**Comment by nlopin** [comment]
Boostnote exports css folder together with html file, so the path is correct. The font issue is well known, there is no good solution of it.

**Comment by tatoosh** [comment]
External CSS are also not a good solution, why not including it in the header?
The image ptioblem also exists.

**Comment by Rokt33r** [comment]
> External CSS are also not a good solution

I agree with it.

**Comment by nlopin** [comment]
Could you explain, why do you think it's bad?
 
I think it's a reasonable solution. There are a lot of styles in those CSS and you probably don't want to have 40KB of styles in your HTML file.

**Comment by Rokt33r** [comment]
Ah, sorry, I misunderstand it.

External CSS is totally fine.

**Comment by Rokt33r** [comment]
@tatoosh I think external assets are completely fine. Why do you think that we should make huge html?

**Comment by tatoosh** [comment]
@Rokt33r
Thanks for the discussion. 
I am using makrdown for large documentation and an Export in a single HTML file could be transered to other people. Also a single file can be stored in a document management system. 
Pandoc exports for example supports that single file html5 Export.

**Comment by Rokt33r** [comment]
I see, but we cannot include fonts. Anyway, I agree with it if it is optional.

**Comment by tatoosh** [comment]
Of course fonts are optional. CSS and images should be included.

**Comment by tatoosh** [comment]
Or u can support Export to script / Export to pandoc?

**Comment by Rokt33r** [comment]
I mean exporting a single html file(with inline script and inline style).

**Comment by tatoosh** [comment]
No, please do not need cloud storage vor makr that optional! The notes should be stored only on our systems. 
Again, please take a look at pandoc, including images and CSS is not a Problem there. I will upload u a sample if that helps?

**Comment by nlopin** [comment]
Can you attach Markdown with local image and HTML made from it via pandoc?

**Comment by tatoosh** [comment]
No, please we not need cloud storage or make that optional! The notes should be stored only on our systems.
Again, please take a look at pandoc, including images and CSS is not a Problem there. I will upload u a sample if that helps?

**Comment by tatoosh** [comment]
@nlopin i will generate it tomorrow and give u the result here.

**Comment by tatoosh** [comment]
@nlopin 
Here is my demo page for u.
I created it with pandoc (over visual studio code extension, could also be a script call).
Images are base64 coded so u will get a working single file export!

```html<img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAABD0AAAJdCAIAAAC``` 

[test.zip](https://github.com/BoostIO/Boostnote/files/1871565/test.zip)

**Comment by Rokt33r** [comment]
I agree that we don't need cloud stuff as long as we just want to include images.

**Comment by nlopin** [comment]
`dataUri` sounds like reasonable solution. I'll do some research to understand is there any restrictions and how does it work with big images.

**Comment by vandijkstef** [comment]
How about some export options? Give the choice to include all CSS in the HTML or make everything external. (Now it's 50/50!). Fonts are definitly optional, and it makes no sense to reference the app on my own computer, suggesting to use the [system default fonts as fallback](https://css-tricks.com/snippets/css/system-font-stack/).

Now, I got here mainly because I would love to have the HTML export, export into a new folder, so the external assets are neatly contained. Having them all external, allows for easier porting to, for example, a website.
Posting under this issue because I think it adds to the discussion, let me know if you'd like a seperate issue.

**Comment by tatoosh** [comment]
Good way, both options would helped me a lot!

**Comment by ndbeals** [comment]
A possible solution to font embedding (or lackthereof) in HTML files would be to also support exporting to [MHTML](https://en.wikipedia.org/wiki/MHTML) files.
